src/rbindlist.c

```diff
@@ -310,10 +312,14 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg)
       if (firsti==-1) { firsti=i; firstw=w; firstCol=thisCol; }
       else {
         if (!factor && !int64) {
-          if (!R_compute_identical(PROTECT(getAttrib(thisCol, R_ClassSymbol)),
-                                   PROTECT(getAttrib(firstCol, R_ClassSymbol)),
-                                   0)) {
-            error(_("Class attribute on column %d of item %d does not match with column %d of item %d."), w+1, i+1, firstw+1, firsti+1);
+          thisClass = PROTECT(getAttrib(thisCol, R_ClassSymbol));
+          firstClass = PROTECT(getAttrib(firstCol, R_ClassSymbol));
+          if (!R_compute_identical(thisClass, firstClass, 0)) {
+            char* thisClassStr = joinCharVec(thisClass, ", ");
+            char* firstClassStr = joinCharVec(firstClass, ", ");
+            error(_("Class attribute on column %d (%s) of item %d does not match with column %d (%s) of item %d."), w+1, thisClassStr, i+1, firstw+1, firstClassStr, firsti+1);
+            free(thisClassStr);
```

Comment:
UNPROTECT() not free()

Reply:
actually, isn't the UNPROTECT() just below this branch targeting the two allocations?

Reply:
nvm